File: server/src/main/java/au/com/codeka/warworlds/server/html/HtmlRequestHandler.java
package au.com.codeka.warworlds.server.html;

import java.io.File;
import java.io.IOException;
import java.util.Map;

import javax.annotation.Nullable;

import au.com.codeka.carrot.CarrotEngine;
import au.com.codeka.carrot.CarrotException;
import au.com.codeka.carrot.Configuration;
import au.com.codeka.carrot.bindings.MapBindings;
import au.com.codeka.carrot.resource.FileResourceLocator;
import au.com.codeka.warworlds.server.handlers.RequestException;
import au.com.codeka.warworlds.server.handlers.RequestHandler;

/**
 * Handler for requests out of the html directory.
 */
public class HtmlRequestHandler extends RequestHandler {
  private static final CarrotEngine CARROT = new CarrotEngine(new Configuration.Builder()
      .setResourceLocator(
          new FileResourceLocator.Builder(new File("data/html/tmpl").getAbsolutePath()))
      .build());

  protected void render(
      String tmplName,
      @Nullable Map<String, Object> data) throws RequestException {
    getResponse().setContentType("text/html");
    getResponse().setHeader("Content-Type", "text/html; charset=utf-8");
    try {
      getResponse().getWriter().write(CARROT.process(tmplName, new MapBindings(data)));
    } catch (CarrotException | IOException e) {
      throw new RequestException(e);
    }
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/html/HtmlRequestHandler.kt
package au.com.codeka.warworlds.server.html

import au.com.codeka.carrot.CarrotEngine
import au.com.codeka.carrot.CarrotException
import au.com.codeka.carrot.Configuration
import au.com.codeka.carrot.bindings.MapBindings
import au.com.codeka.carrot.resource.FileResourceLocator
import au.com.codeka.warworlds.server.handlers.RequestException
import au.com.codeka.warworlds.server.handlers.RequestHandler
import java.io.File
import java.io.IOException

/**
 * Handler for requests out of the html directory.
 */
open class HtmlRequestHandler : RequestHandler() {
  protected fun render(
      tmplName: String?,
      data: Map<String?, Any?>?) {
    response.contentType = "text/html"
    response.setHeader("Content-Type", "text/html; charset=utf-8")
    response.writer.write(CARROT.process(tmplName, MapBindings(data)))
  }

  companion object {
    private val CARROT = CarrotEngine(Configuration.Builder()
        .setResourceLocator(
            FileResourceLocator.Builder(File("data/html/tmpl").absolutePath))
        .build())
  }
}